code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1318,3 +1318,20 @@
 	to_chat(src, span_danger("You shove [target.name] into [name]!"))
 	log_combat(src, target, "shoved", "into [name]")
 	return COMSIG_CARBON_SHOVE_HANDLED
+
+
+// Checks to see how many hands this person has to sign with.
+/mob/living/carbon/proc/check_signables_state()
+	var/obj/item/bodypart/left_arm = get_bodypart(BODY_ZONE_L_ARM)
+	var/obj/item/bodypart/right_arm = get_bodypart(BODY_ZONE_R_ARM)
+	var/empty_indexes = get_empty_held_indexes()
+	if(length(empty_indexes) == 0 || (length(empty_indexes) < 2 && ((!left_arm || left_arm.bodypart_disabled) || (!right_arm || right_arm.bodypart_disabled))))//All existing hands full, can't sign
+		return SIGN_HANDS_FULL // These aren't booleans
+	if((!left_arm || left_arm.bodypart_disabled) && (!right_arm || right_arm.bodypart_disabled))//Can't sign with no arms!
+		return SIGN_ARMLESS
+	if(HAS_TRAIT(src, TRAIT_HANDS_BLOCKED) || HAS_TRAIT(src, TRAIT_EMOTEMUTE))
+		return SIGN_TRAIT_BLOCKED
+	if(handcuffed) // Cuffed, usually will show visual effort to sign
+		return SIGN_CUFFED
+	if(length(empty_indexes) == 1 || (!left_arm || left_arm.bodypart_disabled) || (!right_arm || right_arm.bodypart_disabled)) // One arm gone
+		return SIGN_ONE_HAND
```

Comment:
You duplicate several times in the if statements, specifically (!left_arm || left_arm.bodypart_disabled) and (!right_arm || right_arm.bodypart_disabled) which could be held as variables instead and compared that way, making the if statements more readable.

Reply:
I never realized that I could use variables for or statements, I'll give it a shot